### ISSUE TITLE
Failed to resolve entry for package "react-open-weather". The package may have incorrect main/module/exports specified in its package.json

### ISSUE THREAD

**Issue by SheldonWBM** [farahat80/react-open-weather#126]
Yarn: 4.3.1
React: 18.3.1
Vite: 5.3.1
Vitest: 1.6.0
React-Open-Weather: 1.3.2 to 1.3.6

The error occurs when I run `vitest` after either a fresh `yarn install` or upgrading packages:
`Error: Failed to resolve entry for package "react-open-weather". The package may have incorrect main/module/exports specified in its package.json.`

This version, in `yarn.lock`, of react-open-weather works:
```
"react-open-weather@npm:^1.3.2":
  version: 1.3.2
  resolution: "react-open-weather@npm:1.3.2"
  dependencies:
    "@emotion/styled": "npm:^11.10.0"
    axios: "npm:^0.27.2"
    dayjs: "npm:^1.9.6"
    prop-types: "npm:^15.7.2"
  peerDependencies:
    react: ">=17.0.1"
    react-dom: ">=17.0.1"
  checksum: 10c0/af84cbc56e6a353083492815c5311ab431f11e4ccffcce7382e72661feb1c7ca595ddb9617e3dba936a8a95def31b8453ac22666ff77dfc43f3fb1ac7acbd579
  languageName: node
  linkType: hard
```
The following do not work:
```
"react-open-weather@npm:^1.3.2":
  version: 1.3.6
  resolution: "react-open-weather@npm:1.3.6"
  dependencies:
    "@emotion/styled": "npm:^11.10.0"
    axios: "npm:^1.6.0"
    dayjs: "npm:^1.9.6"
    prop-types: "npm:^15.7.2"
  peerDependencies:
    react: ">=17.0.1"
    react-dom: ">=17.0.1"
  checksum: 10c0/d0ed0dbe6169f5f6153d1ed79fe72a48be2295fed99ed98fb79c575cddb7cd6c398af734fa98a727e4f07a2380bde737930ae5500e42daaef92b66b57d98a21e
  languageName: node
  linkType: hard
```
and
```
"react-open-weather@npm:^1.3.6":
  version: 1.3.6
  resolution: "react-open-weather@npm:1.3.6"
  dependencies:
    "@emotion/styled": "npm:^11.10.0"
    axios: "npm:^1.6.0"
    dayjs: "npm:^1.9.6"
    prop-types: "npm:^15.7.2"
  peerDependencies:
    react: ">=17.0.1"
    react-dom: ">=17.0.1"
  checksum: 10c0/d0ed0dbe6169f5f6153d1ed79fe72a48be2295fed99ed98fb79c575cddb7cd6c398af734fa98a727e4f07a2380bde737930ae5500e42daaef92b66b57d98a21e
  languageName: node
  linkType: hard
```

**Comment by killerfox007** [comment]
I am also running into this issue but cant find a fix

**Comment by SheldonWBM** [comment]
I am just Spitballing: The issue could be precisely what the error states, `"main": "/lib/ReactWeather.js" is incorrect.
In version 1.3.2 (extracting), 
```
.yarn/cache/react-open-weather-npm-1.3.2-5d4d9d026a-10c0/node_modules/react-open-weather/lib/ReactWeather.js
```
In version 1.3.6 (extracting)
```
.yarn/cache/react-open-weather-npm-1.3.6-8210061dcf-d0ed0dbe61/node_modules/react-open-weather/src/js/components/ReactWeather.js
```
There is also a `main.js`
So it was either not built correctly before being published, or the `"main"` in `package.json` needs to be updated.

**Comment by coopernicholas** [comment]
Same error for me

**Comment by farahat80** [comment]
Thanks @SheldonWBM for reporting and investigating, can you confirm the issue is resolved on version 1.3.8?

**Comment by SheldonWBM** [comment]
It is resolved. Thank you @farahat80
